torchsparse/utils/kernel_region.py

```diff
@@ -1,18 +1,32 @@
 import numpy as np
 import torch
+from typing import Union, List, Tuple
 
 __all__ = ['KernelRegion']
 
 
 class KernelRegion:
     def __init__(self,
-                 kernel_size: int = 3,
-                 tensor_stride: int = 1,
-                 dilation: int = 1,
-                 dim=[0, 1, 2]) -> None:
+                 kernel_size: Union[int, List[int], Tuple[int, int, int]] = 3,
+                 tensor_stride: Union[int, List[int], Tuple[int, int, int],
+                                      torch.Tensor] = 1,
+                 dilation: Union[int, List[int], Tuple[int, int, int]] = 1,
+                 dim: List[int] = [0, 1, 2]) -> None:
         self.kernel_size = kernel_size
-        self.tensor_stride = tensor_stride
+        if isinstance(tensor_stride, int):
+            self.tensor_stride = [tensor_stride] * 3
+        elif isinstance(tensor_stride, torch.Tensor):
+            self.tensor_stride = tensor_stride.cpu().view(-1).numpy().tolist()
+        else:
+            self.tensor_stride = tensor_stride
+        if isinstance(dilation, int):
+            dilation = [dilation] * 3
         self.dilation = dilation
+        assert len(self.tensor_stride) == 3, 'Wrong tensor_stride'
+        assert len(self.dilation) == 3, 'Wrong dilation'
+
+        ts = self.tensor_stride
+        d = self.dilation
 
         if not isinstance(kernel_size, (list, tuple)):
```

Comment:
Would it be possible to integrate two cases together by converting `kernel_size` to a tuple?

Reply:
I tend to leave `KernelRegion` related stuff for future PR since I'm not quite sure about the best way to define the kernel region for asymmetric kernels. E.g.:
- `0` to `K-1` or `-(K-1)//2` to `(K-1)//2`
- `[[x, y, z] for x in x_offset for y in y_offset for z in z_offset]` or `[[z, y, x] for x in x_offset for y in y_offset for z in z_offset]`